### PR TITLE
ADR 14: Add Sapphire support

### PR DESCRIPTION
**Pull request by matevz**
This PR:
- removes redundant `sig_context` from `Meta` since it needs to be computed by the hardware wallet anyway
- requires `call.format` to either be `0` or `1`
- adds Sapphire runtime ID on Testnet
- adds references to actual code for all described transactions
- minor styling changes

Preview: https://github.com/oasisprotocol/adrs/blob/matevz/0014/add-sapphire/0014-runtime-signing-tx-with-hardware-wallet.md

### CODE REVIEW

File: 0014-runtime-signing-tx-with-hardware-wallet.md [hunk 1]
```diff
@@ -76,13 +77,17 @@ pub struct Call {
...
 
-If [`format`][runtime-sdk format] equals `0`, the transaction is unencrypted
+If [`format`][runtime-sdk format] is `0`, the transaction is unencrypted
 and it is CBOR-encoded inside the `body` field.
 
-If `format` equals `1`, the transaction is encrypted. In this case `method` is
-empty and `body` contains a CBOR-encoded
-[`CallEnvelopeX25519DeoxysII`][runtime-sdk envelope] which includes the
-encrypted transaction body inside the `data` field.
+If `format` is `1`, the transaction is encrypted.
```
[the remark this answers is not in the document]
Comment (matevz) [reply 0014-runtime-signing-tx-with-hardware-wallet.md:83]: Fair point. I added notice that they should be rejected now.

File: 0014-runtime-signing-tx-with-hardware-wallet.md [hunk 2]
```diff
@@ -682,57 +695,70 @@ In this case the hardware wallet renders the following UI.
...
+field.
 
-`PUBLIC_KEY` is a hex representation of the 32-byte `tx.call.pk` field.
+`ADDRESS` is a hex-encoded address of the smart contract.
 
-`NONCE` is a hex representation of the 15-byte `tx.call.nonce` field.
+### Signing encrypted runtime transactions
 
-### Signing EVM transactions
+Encrypted transactions (`tx.call.format != 0`) contain the call data inside the
```
Comment (kostko) [inline 0014-runtime-signing-tx-with-hardware-wallet.md:730]: This should not be `tx.call.format != 0` but rather `tx.call.format == 1` as you can't be sure that all future formats will be for encryption.

**Review by tjanez — COMMENTED** [review]
Can you break up the changes in 2 commits:
- Other ADR 0014 improvements and updates
- Sapphire support

**Review by tjanez — COMMENTED** [review]
Also update the Change Log

File: 0014-runtime-signing-tx-with-hardware-wallet.md [hunk 3]
```diff
@@ -682,57 +701,70 @@ In this case the hardware wallet renders the following UI.
...
-#### Calling smart contract
-
-In contrast to `contracts.Call`, `evm.Call` would require contract ABI in order
-to extract argument names which are stored in the RLP-encoded transaction on
-the blockchain. We do not support this, so only **blind signing** is performed
-which the user needs to enable first. The UI should be as follows:
+The hardware wallet is not expected to implement any of these decryption
+schemes, neither it is safe to share the ephemeral key with anyone. Instead,
+the user should enable **blind signing** and the hardware wallet should
+show the hash of the encrypted transaction, the public key and the nonce:
```
Comment (kostko) [inline 0014-runtime-signing-tx-with-hardware-wallet.md:743]: How would the wallet be able to show the public key and nonce if it is only given the transaction hash?

Comment (kostko) [reply 0014-runtime-signing-tx-with-hardware-wallet.md:743]: Oh I guess I got confused by "show the hash of the encrypted transaction". So you probably want to say "show the hash of the encrypted call data".

**Review by kostko — APPROVED** [review]

**Review by tjanez — APPROVED** [review]

**Review by nikolaglumac — APPROVED** [review]
Great work @matevz 💯

**Pull request merged**